### PR TITLE
Refactor of tuque, updated constants and doc comments.

### PR DESCRIPTION
**Pull request by MorganDawe**

### CODE REVIEW

File: FedoraRelationships.php [hunk 1]
```diff
@@ -659,7 +693,7 @@ public function get($predicate_uri = NULL, $predicate = NULL, $object = NULL, $t
     if ($type === TRUE) {
       $type = RELS_TYPE_PLAIN_LITERAL;
     }
-    elseif ($type == FALSE) {
+    elseif ($type === FALSE) {
```
Comment (adam-vessey) [inline FedoraRelationships.php]: This is changing behaviour... I'm not really seeing the purpose.

File: FedoraRelationships.php [hunk 2]
```diff
@@ -597,12 +630,13 @@ public function add($predicate_uri, $predicate, $object, $type = RELS_TYPE_URI)
...
    * @param boolean $literal
-   *   Defines if the $object is a literal or not.
+   *   Defines if the $object is a literal or not. Defaults
+   *   to FALSE.
    *
    * @return boolean
    *   TRUE if relationships were removed, FALSE otherwise.
    */
-  public function remove($predicate_uri = NULL, $predicate = NULL, $object = NULL, $literal = FALSE) {
+  public function remove($predicate_uri = NULL, $predicate = NULL, $object = NULL, $literal = OBJ_IS_LITERAL) {
```
Comment (adam-vessey) [inline FedoraRelationships.php]: Really, should just change the `$literal` variable... It just looks like the RDF types weren't properly carried through when they were implemented (including changing the comments).

File: FedoraRelationships.php [hunk 3]
```diff
@@ -471,7 +504,7 @@ protected function internalGet($subject, $predicate_uri = NULL, $predicate = NUL
    * @return boolean
    *   TRUE if relationships were removed, FALSE otherwise.
    */
-  protected function internalRemove($subject, $predicate_uri, $predicate, $object, $literal = FALSE) {
+  protected function internalRemove($subject, $predicate_uri, $predicate, $object, $literal = OBJ_IS_LITERAL) {
```
Comment (adam-vessey) [inline FedoraRelationships.php]: Same deal with `$literal`...

File: FedoraRelationships.php [hunk 4]
```diff
@@ -11,10 +11,22 @@
 define("ISLANDORA_RELS_EXT_URI", 'http://islandora.ca/ontology/relsext#');
 define("ISLANDORA_RELS_INT_URI", "http://islandora.ca/ontology/relsint#");
 
-define("RELS_TYPE_URI", 0);
```
Comment (jordandukart) [inline FedoraRelationships.php:14]: This is going to break things crazily. The INT values correspond to different states when we are creating relationships. It is not just `TRUE` or `FALSE` in this sense as they correspond to the switch here: https://github.com/Islandora/tuque/blob/1.x/FedoraRelationships.php#L297-L316. The docs need to be updated to state that it is in fact not a boolean and perhaps elaborate on the types.

File: RepositoryConnection.php [hunk 5]
```diff
@@ -34,6 +34,8 @@ class RepositoryConnection extends CurlConnection implements RepositoryConfigInt
   public $username;
   public $password;
 
+  const FEDORA_URL = "http://localhost:8080/fedora";
```
Comment (jordandukart) [inline RepositoryConnection.php:37]: This regresses the code. The Fedora URL should not be a constant as this no longer allows us to instantiate connections with repositories that live on other servers besides localhost.

File: RepositoryConnection.php [hunk 6]
```diff
@@ -44,7 +46,7 @@ class RepositoryConnection extends CurlConnection implements RepositoryConfigInt
    * @param string $password
    *   The password to connect with.
    */
-  function __construct($url = 'http://localhost:8080/fedora', $username = NULL, $password = NULL) {
+  function __construct($url = self::FEDORA_URL, $username = NULL, $password = NULL) {
```
Comment (jordandukart) [inline RepositoryConnection.php:49]: See above.

File: FedoraRelationships.php [hunk 7]
```diff
@@ -145,7 +157,7 @@ public function registerNamespace($alias, $uri) {
    *   Determines exiting autoCommit state.
    *   Defaults to TRUE.
    */
-  public function commitRelationships($set_auto_commit = TRUE) {
+  public function commitRelationships($set_auto_commit = AUTO_COMMIT) {
```
Comment (adam-vessey) [inline FedoraRelationships.php]: Why bother using a ~~literal~~ constant for a boolean?

File: FedoraRelationships.php [hunk 8]
```diff
@@ -11,10 +11,22 @@
 define("ISLANDORA_RELS_EXT_URI", 'http://islandora.ca/ontology/relsext#');
 define("ISLANDORA_RELS_INT_URI", "http://islandora.ca/ontology/relsint#");
 
-define("RELS_TYPE_URI", 0);
+define("RELS_TYPE_URI", FALSE);
+define("OBJ_IS_LITERAL", FALSE);
+define("DS_NEW", FALSE);
+define("AUTO_COMMIT", TRUE);
+define("NON_MAGIC_AUTO_COMMIT", TRUE);
```
Comment (adam-vessey) [inline FedoraRelationships.php]: Two constants which map the same value to try to do opposite things is asking for a bad time.

File: FedoraApiSerializer.php [hunk 9]
```diff
@@ -333,14 +336,14 @@ public function getRelationships($request) {
 
     $dom = $this->loadDomDocument($request['content']);
     $xpath = new DomXPath($dom);
-    $results = $xpath->query('/rdf:RDF/rdf:Description/*');
+    $results = $xpath->query(RDF_QUERY);
```
Comment (adam-vessey) [inline FedoraApiSerializer.php]: Why bother using a constant for a string which is only used in one place?

File: FedoraApiSerializer.php [hunk 10]
```diff
@@ -5,6 +5,9 @@
  * something more easily dealt with in PHP.
  */
 
+define("ATTR_NAMESPACE", "http://www.w3.org/1999/02/22-rdf-syntax-ns#");
```
Comment (adam-vessey) [inline FedoraApiSerializer.php]: Should probably be more descriptive, if it's going to be a constant... `RDF_NAMESPACE`, or sommat?

File: RepositoryConnection.php (hunk 6 above)
Comment (adam-vessey) [inline RepositoryConnection.php:49]: This (and this constant) is just the default, yeah? Shouldn't really be an issue... What is kind of weird is having a constant for something which is only used once...

File: RepositoryConnection.php (hunk 5 above)
Comment (adam-vessey) [inline RepositoryConnection.php:37]: This is just used for the default in the constructor... shouldn't really be an issue.

File: FedoraApiSerializer.php (hunk 9 above)
Comment (MorganDawe) [inline FedoraApiSerializer.php]: My thought was it would be easier to edit/find in the future, should it ever need to be, if it were a string defined at the top of the file. Just convenience.

Comment (adam-vessey) [inline FedoraApiSerializer.php]: It becomes that much more difficult to actually read the code though, since we're to be parsing what the query returns on the next few lines... Also, `RDF_QUERY` doesn't really describe the purpose... Kind of overly general.

File: FedoraRelationships.php (hunk 4 above)
Comment (MorganDawe) [inline FedoraRelationships.php:14]: This is still in this file, but reorganized from latest commit.

File: FedoraRelationships.php [hunk 11]
```diff
@@ -330,6 +359,7 @@ protected function internalAdd($subject, $predicate_uri, $predicate, $object, $t
    */
   protected function getXpathResults($xpath_object, $subject, $predicate_uri, $predicate, $object, $type) {
     $xpath = '/rdf:RDF/rdf:Description[@rdf:about="info:fedora/' . $subject . '"]';
+    $type = intval($type);
```
Comment (willtp87) [inline FedoraRelationships.php]: Why intval?

File: FedoraRelationships.php [hunk 12]
```diff
@@ -321,6 +335,21 @@ protected function internalAdd($subject, $predicate_uri, $predicate, $object, $t
   /**
    * This function is used to create an xpath expression based on the input.
    *
+   * Add a new relationship.
```
Comment (willtp87) [inline FedoraRelationships.php]: Copy paste?

File: FedoraApiSerializer.php [hunk 13]
```diff
@@ -333,14 +335,14 @@ public function getRelationships($request) {
 
     $dom = $this->loadDomDocument($request['content']);
     $xpath = new DomXPath($dom);
-    $results = $xpath->query('/rdf:RDF/rdf:Description/*');
+    $results = $xpath->query("/rdf:RDF/rdf:Description/*");
```
Comment (willtp87) [inline FedoraApiSerializer.php]: Why change string type?

File: FedoraRelationships.php [hunk 14]
```diff
@@ -624,7 +656,7 @@ public function remove($predicate_uri = NULL, $predicate = NULL, $object = NULL,
    * @param mixed $type
    *   What the attribute type should be. One of the defined literals beginning
    *   with RELS_TYPE_.  For backwards compatibility we support TRUE as
-   *   RELS_TYPE_PLAIN_LITERAL and FALSE as RELS_TYPE_URI.
+   *   RELS_TYPE_PLAIN_LITERAL and 0 as RELS_TYPE_URI.
```
Comment (willtp87) [inline FedoraRelationships.php]: The old signature was for ==FALSE not ===FALSE; definitely not 0, please revert this change.

File: FedoraRelationships.php (hunk 11 above)
Comment (MorganDawe) [inline FedoraRelationships.php]: I had read there were some backwards compatibility in place (using bool values true/false as int), but that is no longer the case. I am removing this line.

Comment (willtp87) [inline FedoraRelationships.php]: This code is intended to be backwards compatible.  Do you see a way that it is not?
